STYLE.md

```diff
@@ -0,0 +1,453 @@
+
+If data type has only one constructor then this data type name should be the same
+as constructor name (also applies to `newtype`).
+
+```haskell
+data User = User Int String
+```
+
+Field names for record data types:
+
+Choose a reasonable prefix is used such that we can ensure the field name will remain unique throughout the package. Use your discretion.
+
+e.g
+```haskell
+data MempoolEnv m blk = MempoolEnv
+  { meLedger :: LedgerInterface m blk
+  , meLedgerCfg :: LedgerConfig blk
+  , meStateVar :: StrictTVar m (InternalState blk)
+  , meTracer :: Tracer m (TraceEventMempool blk)
+  }
+```
+
+### Comments
+
+#### Punctuation
+
+Write proper sentences; start with a capital letter and use proper
+punctuation.
+
+#### Top-Level Definitions
+
+* Comment top level functions (particularly exported functions)
+as much as possible. If it is not blindingly obvious what a function
+is doing, you should add a comment.
+* Always provide a type signature
+* Use Haddock syntax in the comments.
+* Comment every exported data type.
+
+Function example:
+
+```haskell
+-- | Send a message on a socket. The socket must be in a connected
+-- state. Returns the number of bytes sent. Applications are
+-- responsible for ensuring that all data has been sent.
+send
+  :: Socket      -- ^ Connected socket
+  -> ByteString  -- ^ Data to send
+  -> IO Int      -- ^ Bytes sent
+```
+
+* For functions, the documentation should give enough information to
+apply the function without looking at its definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+  { age :: !Int     -- ^ Age
+  , name :: !String  -- ^ First name
+  }
+```
+
+For fields that require longer comments, format them this way:
+
+```haskell
+data Record = Record
+  { -- | This is a very very very long comment that is split over
+    -- multiple lines.
+    field1 :: !Text
+
+    -- | This is a second very very very long comment that is split
+    -- over multiple lines.
+  , field2 :: !Int
+  }
+```
+
+#### End-of-Line Comments
+
+Separate end-of-line comments from the code with 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser =
+  Parser
+    !Int         -- ^ Current position
+    !ByteString  -- ^ Remaining input
+
+foo :: Int -> Int
+foo n = salt * 32 + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+#### Links
+
+Use in-line links economically.  You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your opinion), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Top-down guideline
+------------------
+
+### LANGUAGE extensions section
+
+Write each `LANGUAGE` pragma on its own line, sort them alphabetically and
+align by max width among them.
```

Comment:
In the below code block, aren't we *not* aligning by width? I think that's the right approach for minimizing semantically-null patches.
```suggestion
```

Reply:
:100: